### DISCUSSION TITLE
Add ability to see that Trivy doesn't support detecting vulnerabilities in the found OS for `json` format

### DISCUSSION THREAD

**Discussion by DmitriyLewen** [aquasecurity/trivy#6971] [Ideas]
### Description

Trivy shows information about unsupported OS only in log message (e.g. `2024-05-04T08:27:07Z	WARN	Unsupported os	family="fedora"`).
But if log is supressed (`--quiet` flag) - users don't have option to understand that Trivy doesn't support detecting vulnerabilities in the found OS.

As solution we can add new `bool` field (e.g. `vulnScanningSupported`) into [OS](https://github.com/aquasecurity/trivy/blob/e493fc931a2e55888a84b7db51f3dd6f01028c9b/pkg/fanal/types/artifact.go#L9-L16) struct.
Or we can use `string` field (`supported`, `unsupported` and `disabled`) to avoid confusing users when vuln vulnerability scanning is disabled.

### Target

Container Image

### Scanner

Vulnerability

**Comment by knqyf263** [comment]
We need a "supported" flag both for SBOM and vulnerabilities.

**Comment by Bah-bah** [comment]
Any chance it will be available in the next release?

**Comment by badgerinspace** [comment]
This would be extremely helpful for companies that triage vulnerabilities based on the JSON files alone. As is, there's no indication in that workflow that no vulnerabilities were detected because the OS isn't supported.

**Comment by DmitriyLewen** [comment]
Created #8256 for this task
